python/ray/data/_internal/util.py

```diff
@@ -421,3 +425,38 @@ def capitalize(s: str):
         Capitalized string with no underscores.
     """
     return "".join(capfirst(x) for x in s.split("_"))
+
+
+def _df_to_block(df: "pandas.DataFrame") -> "Block[ArrowRow]":
+    from ray.data.block import BlockAccessor, BlockExecStats
+
+    stats = BlockExecStats.builder()
+    import pyarrow as pa
+
+    block = pa.table(df)
+    return (
+        block,
+        BlockAccessor.for_block(block).get_metadata(
+            input_files=None, exec_stats=stats.build()
+        ),
+    )
+
+
+def _ndarray_to_block(ndarray: np.ndarray) -> "Block[np.ndarray]":
+    from ray.data.block import BlockAccessor, BlockExecStats
+
+    stats = BlockExecStats.builder()
+    block = BlockAccessor.batch_to_block(ndarray)
+    metadata = BlockAccessor.for_block(block).get_metadata(
+        input_files=None, exec_stats=stats.build()
+    )
+    return block, metadata
+
+
+def _get_metadata(table: Union["pyarrow.Table", "pandas.DataFrame"]) -> "BlockMetadata":
```

Comment:
let's remove `_` as they are utility methods, and I think a explicit naming would be helpful:

```
def get_table_block_metadata()
```